### DISCUSSION TITLE
Dedicated st2actiorunner for specific tasks

### DISCUSSION THREAD

**Discussion by ppanchal10** [StackStorm/st2#5562] [Ideas]
Hi, We are using Stackstorm version 3.1.0 and I have question around st2actionrunner which is responsible for scheduling & executing incoming requests through RabbitMQ. We are running multiple instances for st2actionrunner for HA & better performance but we would like achieve more granularity by creating dedicated st2actionrunner for specific tasks. Based on my understanding there is a queue which st2actionrunner (or its components) listens and execute any incoming request on that queue. Can we create the custom queues based on tasks and have dedicated st2actionrunner associated to that queue. Please advise.

**Comment by arm4b** [comment]
This feature request was discussed in the [Feature req: Partitioning actions to execute on separate nodes #3096 ](https://github.com/StackStorm/st2/issues/3096).

We're open to accept contributions for this kind of functionality to stackstorm.

**Reply by ppanchal10** [reply]
Thank you @armab for quick response. I would love to contribute, I am very new Stackstorm and project in general, would you please share some information on how to start design discussion, PR submission  and actual implementation.

**Reply by arm4b** [reply]
You can start gathering some feedback and socializing in the https://github.com/StackStorm/st2/issues/3096 issue itself (it has some tips on where to start), or just here. Another good example is: https://github.com/StackStorm/st2/discussions/5373

You can find the contribution guidelines at https://docs.stackstorm.com/development/index.html and https://github.com/StackStorm/st2/blob/master/CONTRIBUTING.rst

Also join the `#development` Channel in the StackStorm community Slack https://stackstorm.com/#community.
